code/obj/item/mops_cleaners.dm

```diff
@@ -54,11 +54,16 @@ WET FLOOR SIGN
 	desc = "A cleaner bottle jury-rigged to synthesize space cleaner."
 	icon_state = "cleaner_robot"
 
+	New()
+		..()
+		processing_items.Add(src)
```

Comment:
```suggestion
	New()
		..()
		processing_items.Add(src)

	disposing()
		..()
		processing_items.Remove(src)	
```
@pali6 Sanity check.

Reply:
there's no `disposing()` in https://github.com/goonstation/goonstation/blob/master/code/modules/chemistry/tools/robospray.dm, which is what I based this on

Reply:
Yeah it should absolutely be removed in disposing.

Reply:
Honestly I think I'd prefer to have the processing_items registration happen in `afterattack` where the spraying itself happens:
```
afterattack(atom/A, mob/user)
		. = ..()
		if (src.reagents.total_volume < 25)
      processing_items |= src
```
and then have
```
		if (src.reagents.total_volume < 25)
			src.reagents.add_reagent("cleaner", 1)
    else
      processing_items.Remove(src)
```
in `process`. That way you avoid it `processing` pointlessly when full.